### ISSUE TITLE
Tooltip is not working under conditional statements

### ISSUE THREAD

**Issue by jigneshchudasama05** [sir-dunxalot/ember-tooltips#17]
```
<div>
{{------This is working well---}}
 <span class="fa fa-check text-green cursor-pointer has-tooltip form-code-validation absolute" data-tooltip-content={{tooltipData}}></span>

{{---This is not working---}}
{{#if isValidCode}}
            <span class="fa fa-check text-green cursor-pointer has-tooltip form-code-validation absolute" data-tooltip-content={{tooltipData}}></span>
          {{else}}
            <span class="fa fa-times text-red cursor-pointer has-tooltip form-code-validation absolute" data-tooltip-content={{tooltipData}}></span>
          {{/if}}
</div>
```

**Comment by sir-dunxalot** [comment]
If you are using the `data-<x>` attributes you will need to call, in the parent view/component, `this.renderChildTooltips()` everytime `isValidCode` changes. For example:

```js
// app/components/some-component.js

import Ember from 'ember';

export default Ember.Component.extend({

  renderTooltips: Ember.on('didInsertElement', Ember.observes('isValidCode', function() {
      Ember.run.scheduleOnce('afterRender', this, function() {
        this.renderChildTooltips(); // Voila!
      });
  })),

});
```

**Comment by sir-dunxalot** [comment]
Let me know if that doesn't solve your issue.

**Comment by jigneshchudasama05** [comment]
Thanks !! :) 
will try this and let you know

**Comment by sir-dunxalot** [comment]
Is this still an issue for you?

**Comment by alvinvogelzang** [comment]
@sir-dunxalot there's a typo in your example. `Ember.observes` should be `Ember.observer`. Beside the typo your example is working but using observers is not a very elegant solution I think.

**Comment by sir-dunxalot** [comment]
Well spotted! I agree - observers suck. The best option would probably be using Ember's new component lifecycle hooks or something like that. Do you have any ideas?

**Comment by alvinvogelzang** [comment]
Nice suggestion. You can replace the observer and `didInsertElement()` with `didUpdateAttrs()`. This is working for me:

```
// app/components/some-component.js

import Ember from 'ember';

export default Ember.Component.extend({

  didUpdateAttrs: function() {
    this.renderChildTooltips(); 
  }
});
```

**Comment by sir-dunxalot** [comment]
Very interesting - I've never had much success getting the new hooks to work so that's cool. Which version of Ember are you using?

**Comment by alvinvogelzang** [comment]
1.13.10. It should work form 1.13.0 I think: http://emberjs.com/blog/2015/05/24/another-two-oh-status-update.html#toc_new-lifecycle-hooks

**Comment by sir-dunxalot** [comment]
I believe this has been fixed with release `1.0.0`: https://github.com/sir-dunxalot/ember-tooltips/releases/tag/1.0.0

If that is incorrect we can reopen this issue.
